docs/development/buildenvironments.rst

```diff
@@ -82,3 +82,22 @@ DOCKER_VERSION
     Version of the API to use for the Docker API client.
 
     Default: :djangosetting:`DOCKER_VERSION`
+
+
+Local development
+-----------------
+
+On Linux development environments, it's likely that your UID and GID do not
+match the ``docs`` user that is set up as the default user for builds. In this
+case, it's necessary to make a new image that overrides this user::
+
+    cd contrib/
+    ./docker_build.sh latest
+
+This will create a new image, ``readthedocs/build-dev:latest``. To build a
+different image, you can instead specify a version to build::
+
+    cd contrib/
+    ./docker_build.sh 5.0
+
+This will create a new image, ``readthedocs/build-dev:5.0``.
```

Comment:
I got rid of the directions to retag the image as `readthedocs/build:latest` for a couple reasons:

* It's a cyclical name -- this image uses `readthedocs/build:latest` as the base. The next time you try to build a `readthedocs/build-dev:latest`, this image will be used as the base, as so on
* We can probably automate it without user intervention

A couple of options here:

* Automate the image names in `settings.dev` -- iterate over `DOCKER_IMAGE_SETTINGS` and change the keys to `readthedocs/build-dev:latest` etc. The issue with this method is how do we detect *when* we should change the key names? I don't want to have to test docker, but testing if `readthedocs/build:*` are images in docker is the most explicit way. Another option is maybe a setting? `DOCKER_USE_DEV_IMAGES` or whatever.
* Automate tagging as part of the docker_build.sh script and just overwrite the image. You'll have to remember to remove `readthedocs/build:latest` before you build the new latest

@humitos thoughts here?

Reply:
Oh, maybe it's inferring too much, but a property setting in `settings.dev` that is similar to:

```python
if os.geteuid() != 1005 and os.getegid != 1005 and sys.platform == 'linux':
    # replace all the keys in DOCKER_IMAGE_SETTINGS
```

:man_shrugging:

Reply:
Or we can just leave this as a manual step, I have my images as `readthedocs/build-dev` in my local_settings.py file

Reply:
Yeah, we need a way to make this setting optional. I don't want the setting to be autodetected and it picks the dev image that maybe I don't want. I prefer marking something like `USE_DEV_DOCKER=True` as you mentioned.

Reply:
I've added a config option `DOCKER_USE_DEV_IMAGES` for `settings.dev`, which can be set in `local_settings.py`. This will patch the image names.